lib/session.js

```diff
@@ -56,6 +56,13 @@ module.exports = class Session {
     this[persistedHash] = this[hash]()
   }
 
+  options (opts) {
+    if (Object.keys(opts).length) {
+      this[cookieOptsKey] = opts
+      Object.assign(this.cookie, this[cookieOptsKey])
```

Comment:
So we are merging to `this.cookie` but directly assigning to `this[cookieOptsKey]`, aren't we losing its other properties with that direct assignment?

[`.regenerate`](https://github.com/fastify/session/pull/260/files#diff-1f2570998a9bf80486f544bb34c3bbf16e1af1755a37ac836eabbd82c6be4fb4R77-R83) will forget its `maxAge` if we modify `path` for instance, no?

Why not:

```suggestion
      Object.assign(this[cookieOptsKey], opts)
      this.cookie = this[cookieOptsKey]
```

Reply:
> So we are merging to this.cookie but directly assigning to this[cookieOptsKey], aren't we losing its other properties with that direct assignment?

I suppose you're right, better to merge old cookieOptsKey with newer ones, regarding setting direct to this.cookie I don't think it's a good idea since old params will be lost and we don't want to set maxAge but to use [cookie setter](https://github.com/fastify/session/blob/master/lib/cookie.js#L45), for example:

cookie before:
```
{
  path: '/',
  secure: true,
  sameSite: 'strict',
  domain: null,
  httpOnly: true,
  partitioned: undefined,
  _expires: null,
  originalMaxAge: null
}
```
cookie after:
```
{
  secure: true,
  httpOnly: true,
  sameSite: 'strict',
  maxAge: 2592000000
}
```

Reply:
I created this test that fails currently (notice the usage of `regenerate`):

```js
test('#260', t => {
  t.plan(11)

  const fastify = Fastify()
  fastify.register(fastifyCookie)
  fastify.register(fastifySession, {
    ...DEFAULT_OPTIONS,
    cookie: {
      secure: false,
      maxAge: 42,
      path: '/'
    }
  })

  fastify.post('/', (request, reply) => {
    request.session.set('data', request.body)
    request.session.options({ maxAge: 1000 * 60 * 60 }) // maxAge updated to 1 hour

    reply.send('hello world')
  })

  t.teardown(fastify.close.bind(fastify))

  fastify.get('/', async (request, reply) => {
    const data = request.session.get('data')
    await request.session.regenerate()

    if (!data) {
      reply.code(404).send()
      return
    }

    reply.send(data)
  })

  fastify.inject({
    method: 'POST',
    url: '/',
    payload: {
      some: 'data'
    }
  }, (error, response) => {
    t.error(error)
    t.equal(response.statusCode, 200)
    t.ok(response.headers['set-cookie'])
    const { expires, path } = response.cookies[0]
    t.equal(expires.toUTCString(), new Date(Date.now() + 1000 * 60 * 60).toUTCString())
    t.equal(path, '/')

    fastify.inject({
      method: 'GET',
      url: '/',
      headers: {
        cookie: response.headers['set-cookie']
      }
    }, (error, response) => {
      t.error(error)
      t.equal(response.statusCode, 200)
      t.same(JSON.parse(response.payload), { some: 'data' })
      t.ok(response.headers['set-cookie'])
      const { expires, path } = response.cookies[0]
      t.equal(expires.toUTCString(), new Date(Date.now() + 1000 * 60 * 60).toUTCString()) // This will fail
      t.equal(path, '/')
    })
  })
})
```

We need this:

```suggestion
      Object.assign(this[cookieOptsKey], opts)
      this.cookie = new Cookie(this[cookieOptsKey], this[requestKey])
```